src/Tribe/Views/V2/Views/List_View.php

```diff
@@ -190,6 +190,15 @@ protected function get_upcoming_url( $canonical = false, $page = 1 ) {
 
 		$upcoming->order_by( '__none' );
 
+		/**
+		 * Check whether the Main Events Page is set as the homepage.
+		 * 
+		 * @since TBD
+		 */
+		if ( ! tribe_is_events_front_page() ) {
```

Comment:
I can't argue it works, but I have some concern about the logic here 🤔

Reply:
I actually have A PR for a (IMO) better solution here: https://github.com/the-events-calendar/the-events-calendar/pull/3879